### DISCUSSION TITLE
Twig version in use

### DISCUSSION THREAD

**Discussion by brendanpetty** [getgrav/grav#3681] [General]
It looks like Grav still uses Twig 1.x while Twig v 3 has been around for quite some time and is better set up for running on PHP 8. Are there options for using a different Twig version, or a timeline for shifting to this?

**Comment by brendanpetty** [comment]
@rhukster are you able to advise on this please?

**Reply by rhukster** [reply]
Really it comes down to compatibility.  Even version twig2 requires plugin changes. Existing plugins that are not compatible will cause the site to break.  We can easily fix our own but 3rd party plugins and even custom stuff people have done for sites are the problem. 

It really can only be addressed in a major grav update (like a grav 2.0) that breaks compatibility and is not an automatic update.

**Reply by brendanpetty** [reply]
Ok, that's helpful to know
